### DISCUSSION TITLE
Force Tasmota to read sensor (pzem) - BERRY

### DISCUSSION THREAD

**Discussion by mmmmm009** [arendst/Tasmota#20904] [Q&A]
Hi,

I have a pzem-004t power meter connected to ESP32 devkitc v4 pins:
A0 GPIO32 - PZEM0XX Tx
A0 GPIO33 - PZEM016 Rx

1. Is that mean that tasmota automatically uses hardware uart? 
pinout: 
![ESP32-DEV-KIT-DevKitC-v4-pinout-mischianti](https://github.com/arendst/Tasmota/assets/72743330/69585dbd-8911-4449-b605-04a1ed197250)

I've checked (using the following rule):
```
Rule1 ON ENERGY#Voltage DO publish stat/sensor/Voltage %value% ENDON
Rule1 4
Rule1 1
```
that tasmota reads this sensor every...~800ms (cmd in tasmota published every 800ms)
and now I'd like to force tasmota to do it more often.

With the following driver, do I force tasmota to get new readings or I just copy stored value to Volt1?
(do I force tasmota (every 250ms) to send a new query via uart?)

THX!

```
import json

var Volt1=0

class my_driver

def every_250ms()
var sensors = json.load(tasmota.read_sensors()) 
Volt1 = (sensors['ENERGY']['Voltage'])
end

end
tasmota.add_driver(my_driver())                     

```

**Comment by s-hadinger** [comment]
With the code above you do force each sensor to read back values.

**Comment by mmmmm009** [comment]
Thx. 

And do tasmota uses hardware uart in my case?

**Reply by s-hadinger** [reply]
It depends on the driver. I suppose yes in your case. It's easy to try and see if you get different readings

**Reply by mmmmm009** [reply]
I don't get the above :/

In any other case... when I use:
IO GPIO16        (hardware rx)
IO GPIO17        (hardware tx)
will tasmota use hardware uart automatically?
![ESP32-DEV-KIT-DevKitC-v4-pinout-mischianti](https://github.com/arendst/Tasmota/assets/72743330/cfa5197c-cacd-498e-a890-035fc693b527)

**Reply by barbudor** [reply]
Esp32 has 3 hardware uart that can be used on ANY gpios
On ESP32, tasmota only use hardware uarts

**Reply by mmmmm009** [reply]
Thx Guys!
